vividus-extension-selenium/src/main/java/org/vividus/steps/ui/GenericWaitSteps.java

```diff
@@ -169,6 +230,15 @@ public void doesElementByLocatorExistsForDuration(Locator locator, Duration dura
         });
     }
 
+    private WebElement findWebElementAndAssertElementExists(SearchContext context, Locator locator)
+    {
+        WebElement webElement = baseValidations
+                .assertElementExists(ELEMENT_TO_VALIDATE_EXISTENCE, context, locator)
+                .orElseThrow(() -> new NoSuchElementException(String.format(NO_SUCH_ELEMENT_EXEPTION,
+                        locator.toHumanReadableString())));
```

Comment:
why is it needed to throw `NoSuchElementException`?

Reply:
updated